src/libaktualizr/uptane/fetcher.cc

```diff
@@ -119,23 +109,19 @@ bool Fetcher::fetchVerifyTarget(const Target& target) {
       if (target.hashes().empty()) {
         throw Exception("image", "No hash defined for the target");
       }
-      bool retry = false;
+      HttpResponse response;
       do {
-        retry = false;
-        HttpResponse response =
-            http->download(config.uptane.repo_server + "/targets/" + Utils::urlEncode(target.filename()),
-                           DownloadHandler, &ds, ds.downloaded_length);
-        if (!response.isOk() && !pause_) {
-          if (response.curl_code == CURLE_WRITE_ERROR) {
-            throw OversizedTarget(target.filename());
-          }
-          throw Exception("image", "Could not download file, error: " + response.error_message);
-        }
-        if (pause_) {
-          std::lock_guard<std::mutex> lock(*pause_mutex_);
-          retry = true;
+        checkPause();
+        retry_ = false;
+        response = http->download(config.uptane.repo_server + "/targets/" + Utils::urlEncode(target.filename()),
+                                  DownloadHandler, &ds, ds.downloaded_length);
+      } while (retry_);
+      if (!response.isOk()) {
+        if (response.curl_code == CURLE_WRITE_ERROR) {
+          throw OversizedTarget(target.filename());
         }
-      } while (retry);
+        throw Exception("image", "Could not download file, error: " + response.error_message);
+      }
```

Comment:
This cleanup is definitely an improvement, but I think we are losing our retry functionality. We currently retry downloads at a high level if they fail for basically any reason. I think that's worth keeping. Originally, I think we gave it three tries before failing.

Reply:
Thanks! I forgot to look into sotauptaneclient.cc. The retry indeed will not work. But does it work now? The code on master throws in exactly the same way.

Reply:
My bad, the actual retry logic is in `SotaUptaneClient::downloadImage`. All is well, sorry!